### PR TITLE
Remove support for old style repositories

### LINKED ISSUE #386: Remove support for old type repostitories
Triggered by #385. The deprecation message is there for a while, we can safely remove support for it.

```
2019-02-14 09:50:20,632 cekit        WARNING  The way of defining repository 'jboss-os' is deprecated. Convert it to an URL based repository object. Consult Cekit docs, for more details.
```

Old repositories are defined like this:

```yaml
packages:
  repositories:
    - jboss-os
```

### PR DESCRIPTION
**Pull request by goldmann**
This removes support for repositories defined this way:

```yaml
repositories:
  - repo-foo
  - repo-bar
```

and defined in the CEKit configuration file.

Fixes #386

### CODE REVIEW

File: cekit/descriptor/packages.py [hunk 1]
```diff
@@ -103,13 +105,14 @@ class Repository(Descriptor):
     """
 
     def __init__(self, descriptor):
-        # we test parameter is not dict as there is no easy way how to test
-        # if something is string both in py2 and py3
-        if not isinstance(descriptor, dict):
-            descriptor = self._create_repo_object(descriptor)
+        # if not isinstance(descriptor, dict):
+        #     raise CekitError(
+        #         "Invalid content found as the repository definition: {}".format(descriptor))
```
Comment (rnc) [inline cekit/descriptor/packages.py]: Commented out code?

**Comment by rnc** [comment]
This doesn't need any docs update does it?

**Comment by goldmann** [comment]
No, it was already removed from docs some time ago, even from stable.

File: cekit/descriptor/packages.py (hunk 1 above)
Comment (goldmann) [reply cekit/descriptor/packages.py]: Good catch! Fixed now.

**Pull request merged**